lib/krane/kubernetes_resource.rb

```diff
@@ -350,13 +355,20 @@ def sensitive_template_content?
     end
 
     def server_dry_runnable_resource?
-      self.class::SERVER_DRY_RUNNABLE
+      self.class::SERVER_DRY_RUNNABLE && @definition.dig('metadata', 'generateName').blank?
     end
 
     def server_dry_run_validated?
       @server_dry_run_validated
     end
 
+    def use_generated_name(instance_data)
+      @name = instance_data.dig('metadata', 'name')
+      @definition['metadata']['name'] = @name
+      @definition['metadata'].delete('generateName')
+      @file = nil
```

Comment:
Is this to force a new write of the updated definition on subsequent calls to file_path? If so, why not `@file = create_definition_tempfile` right here inline?

Reply:
Good idea, let's do that